engine/table/src/main/java/io/deephaven/engine/table/impl/select/analyzers/SelectAndViewAnalyzer.java

```diff
@@ -289,8 +314,9 @@ public static SelectAndViewAnalyzerWrapper create(
                                 rowRedirection, underlyingSource, rowSet.intSize());
                     }
                     maybeCreateAlias.accept(scs);
-                    analyzer = analyzer.createLayerForSelect(updateGraph, rowSet, sc.getName(), sc, scs,
-                            underlyingSource, distinctDeps, mcsBuilder, rowRedirection != null, false, false);
+                    final String name = sc.getName();
+                    analyzer.addLayer(new SelectColumnLayer(updateGraph, rowSet, analyzer, name, sc, scs,
```

Comment:
Passing `analyzer` in concerns me a bit. I wonder if the fields the layers want should be moved to a context?